playground-common/playground-plugin/src/main/kotlin/androidx/playground/PlaygroundPlugin.kt

```diff
@@ -13,38 +13,39 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
- package androidx.playground
+
+package androidx.playground
 
 import org.gradle.api.Plugin
 import org.gradle.api.initialization.Settings
 
-class PlaygroundPlugin implements Plugin<Settings> {
-    void apply(Settings settings) {
-        settings.apply(plugin: "playground-ge-conventions")
-
-        settings.extensions.create("playground", PlaygroundExtension, settings)
-
+class PlaygroundPlugin : Plugin<Settings> {
+    override fun apply(settings: Settings) {
+        settings.apply(mapOf("plugin" to "playground-ge-conventions"))
+        settings.extensions.create("playground", PlaygroundExtension::class.java, settings)
         validateJvm(settings)
     }
 
-    def validateJvm(Settings settings) {
+    private fun validateJvm(settings: Settings) {
         // validate JVM version to print an understandable error if it is not set to the
         // required value (11)
-        def jvmVersion = System.getProperty("java.vm.specification.version")
+        val jvmVersion = System.getProperty("java.vm.specification.version")
         if (jvmVersion != "11") {
```

Comment:
Could just make this a check with a string template:

```
check(jvmVersion == "11") {
    """
    AndroidX build must be invoked with JDK 11.
    ${
        if (settings.gradle.startParameter.projectProperties.containsKey("android.injected.invoked.from.ide")) {
            "Make sure to set the gradle JDK to JDK 11 in the project settings. (File -> Other Settings -> Default Project Structure)"
        } else {
            "Make sure your JAVA_HOME environment variable points to Java 11 JDK."
        }
    }
    Current version: $jvmVersion
    Current JAVA HOME: ${System.getProperty("java.home")}
    """.trimIndent()
}
```

At the very least I think we should add `.trimIndent()` / `.trimMargin()` to cut down the extra whitespace.

Reply:
Done